packages/pigeon/lib/swift_generator.dart

```diff
@@ -667,20 +676,25 @@ String _camelCase(String text) {
   return pascal[0].toLowerCase() + pascal.substring(1);
 }
 
-String _castForceUnwrap(String value, TypeDeclaration type, Root root) {
-  final String forceUnwrap = type.isNullable ? '' : '!';
-  final String castUnwrap = type.isNullable ? '?' : '';
+String _castForceUnwrap(String value, TypeDeclaration type, Root root,
+    {String? variableName}) {
   if (isEnum(root, type)) {
-    final String nullableConditionPrefix =
-        type.isNullable ? '$value == nil ? nil : ' : '';
-    return '$nullableConditionPrefix${_swiftTypeForDartType(type)}(rawValue: $value as! Int)$forceUnwrap';
+    if (type.isNullable) {
+      return '''
+nil
+    let ${variableName}EnumVal: Int? = ${_castForceUnwrap(value, const TypeDeclaration(baseName: 'Int', isNullable: true), root)}
+    if let ${variableName}RawValue = ${variableName}EnumVal {
+      $variableName = ${_swiftTypeForDartType(type)}(rawValue: ${variableName}RawValue)!
+    }''';
```

Comment:
I'm going to pull this out into a new method to make this less weird and fragile